02_housing/2a_affordable_available_place.qmd

```diff
@@ -0,0 +1,253 @@
+               names_pattern = "(.+?(?=_[^_]+$))(_[^_]+$)", # this creates two columns - "share_affordable_XXAMI" and "_owner/_renter/_all"
+               values_to = "value") %>% 
+  # pivot_wider again so that each share_affordable by AMI is it's own column with subgroups as rows
+  pivot_wider(
+    names_from = available, 
+    values_from = value
+  ) %>% 
+  # clean subgroup names and add subgroup type column 
+  # remove leading underscore and capitalize words
+  mutate(subgroup = str_remove(subgroup, "_") %>% str_to_title(),
+         subgroup_type = "tenure") %>% 
+  # join with affordable values for data quality flag 
+  left_join(affordable_2022_subgroup) %>% 
+  # create a share affordable available quality flag
+  # THIS IS THE SAME AS THE SHARE AFFORDABLE QUALITY FLAG
+  # BECAUSE BOTH ARE BASED ON THE UNDERLYING HOUSEHOLD SAMPLE SIZE AND PUMA CROSSWALK
+  # rename the quality flag columns to be share_affordable_available_quality instead of share_afordable_quality
+  mutate(across(matches("share_.*quality$"),
+                list(share_affordable_available = ~.), 
+                .names = "share_affordable_available_{str_remove(.col, 'share_affordable_')}"), 
+         # subpress counties with too small of sample size
+         across(matches("share_affordable_available.*ami$"), \(x) if_else(is.na(get(cur_column() %>% paste0("_quality"))), NA, x))) %>% 
+  # arrange columns and drop unneeded ones
+  select(year, state, place, subgroup_type, subgroup, matches("available_30"),  matches("available_50"), matches("available_80"))
+
+# export our file as a .csv
+write_csv(available_2022_subgroup, here::here("02_housing/data/available_2022_subgroups_city.csv"))  
+```
+
+## (5b) overall file
+
+```{r}
+# combine with share_affordable file 
+affordable_2022_overall <- read_csv(here::here("02_housing/data/housing_2022_city.csv")) %>%
+  select(state, place, matches("quality"))
+
+# keep what we need
+available_2022_overall <- available_2022_subgroup %>% 
+  filter(subgroup == "All") %>% 
+  left_join(affordable_2022_overall) %>% 
+   # create a share affordable available quality flag
+  # THIS IS THE SAME AS THE SHARE AFFORDABLE QUALITY FLAG
+  # BECAUSE BOTH ARE BASED ON THE UNDERLYING HOUSEHOLD SAMPLE SIZE AND PUMA CROSSWALK
+  mutate(across(matches("share_.*quality$"),
+                list(share_affordable_available = ~.), 
+                .names = "share_affordable_available_{str_remove(.col, 'share_affordable_')}"), 
+         # subpress counties with too small of sample size
+         across(matches("share_affordable_available.*ami$"), \(x) if_else(is.na(get(cur_column() %>% paste0("_quality"))), NA, x))) %>% 
+    # arrange columns and drop unneeded ones
+  select(year, state, place, matches("available_30"),  matches("available_50"), matches("available_80"))
+
+
+# export our file as a .csv
+write_csv(available_2022_overall, here::here("02_housing/data/available_2022_city.csv")) 
```

Comment:
The final data is available_2022_overall, correct? My understanding is that it should have 6 variables (3 metrics plus 3 quality flags). This has 3 metrics and 6 quality flags. Perhaps @cdsolari can weigh in as I'm not 100% certain about this.

Reply:
@rogin123 @rpitingolo it looks like the available_2022_city file for the overall values is borrowing the format of the subgroup file. It should only be one census place per row, and it shouldn't have the subgroup_type or subgroup variables in that one. It should look more like your county version of that file (though see below because there are issues there too). I think this will be addressed it you limit it to just the All rows, but note that you have some strange things going on with your all rows because you have 3 per place, and not all of them have the same values. If you have them on multiple rows, I'd expect them to be duplicates and you can just remove the renter or owner subgroups and deduplicate on the all and then remove the two variables to indicate subgroup. But, I'm not sure what value is the correct "all" value because some are identical while others are different. The main difference seems to be some where you have more values that are NA. Let me know if I can help iron this out. I also see a similar problem in the county file. It looks like you kept the "all" rows and removed the subgroup_type and subgroup variables, but you still have 3 rows per county, and they aren't showing the same for each. I have a feeling you have code that maybe progressively gets more restrictive, but instead of applying it to the same row, it creates more rows that is each progressively more restrictive. I'm happy to show  you what I'm seeing.

Reply:
@cdsolari regarding the 3 rows per place issue. How are you looking at the data? Are you looking at the CSVs pushed to github? Or looking at the dataframe in R Studio? This was an issue I previously flagged for Amy that I think is now resolved so want to make sure we are all looking at the same thing.

Reply:
@rpitingolo You're right, I was looking at an older version of the file. Thanks for catching that!